### PR TITLE
Add default configuration files for v22.4

### PR DESCRIPTION
**Pull request by HashidaTKS**
This is a same patch as #30 (cherry-picked).

* Add default configuration files

The default configuration files are Common.txt, TrustedDomains.txt, UnsafeDomains.txt and UnsafeFiles.txt in `{app}\DefaultConfig`. Those files are installed when they are located in the "DefaultConfig" folder that exits in the same folder to installer like below.

```
│  FlexConfirmMailSetup-22.4.0.exe
│
└─DefaultConfig
        Common.txt
        TrustedDomains.txt
        UnsafeDomains.txt
```

The configurations in the policy configuration and the default configuration files for trusted domains, unsafe domains and unsafe files are fixed, they can't modified by user configuration files. The priority of common configurations is as following.

User file configurations > Default file configurations > Policy configurations

## Tests

Prerequisites:

* Default Config:  `C:\Program Files\FlexConfirmMail\DefaultConfig\*.txt`
* User Config: `%UserAppData%\FlexConfirmMail\*.txt`
* Registry Config: Config specified by GPO
* "Merge" means:
  * For Trusted Domains, Unsafe Domains, Unsafe Files
    * All configurations of the policy configuration, the default configuration and the user configuration are used.
    *  The configurations in the policy configuration and the default configuration files are fixed, they can't modified by user 
  * For Common Config
    *  If a setting is modified from default value, the highest priority config are used. The priority is as following.
    * User file configurations > Default file configurations > Policy configurations.

Test:

* [x] Confirm that `DefaultConfig/*.txt` is installed to `C:\Program Files\FlexConfirmMail\DefaultConfig\*.txt`.
* [x] Confirm that default config is used if there is only default config (remove other settings if exists).
* [x] Confirm that registry config is used if there is only registry config  (remove other settings if exists).
* [x] Confirm that user config is used if there is only user config  (remove other settings if exists).
* [x] Confirm that registry config and default config are merged if they are specified.
* [x] Confirm that default config and user config are merged if they are specified.
* [x] Confirm that registry config and user config are merged if they are specified.
* [x] Confirm that all configs are merged if they are specified.

### CODE REVIEW

**Comment by HashidaTKS** [comment]
I don't test this yet.
I will do test this.

**Comment by HashidaTKS** [comment]
I have tested and it worked fine.

File: Dialog/ConfigDialog.xaml.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ namespace FlexConfirmMail.Dialog
     public partial class ConfigDialog : Window
     {
         Config _config = new Config();
-        Config _default = new Config();
+        Config _default = new Config();
```
Comment (piroor) [inline Dialog/ConfigDialog.xaml.cs:14]: ここは単に改行コードが変わっただけでしょうか？

File: Dialog/ConfigDialog.xaml.cs [hunk 2]
```diff
@@ -24,7 +24,10 @@ public ConfigDialog()
             {
                 _default = Loader.LoadFromReg(RegistryPath.DefaultPolicy);
                 _config.Merge(_default);
-            }
+            }
+            Config defaultWithFile = Loader.LoadFromDir(StandardPath.GetDefaultConfigDir());
```
Comment (piroor) [inline Dialog/ConfigDialog.xaml.cs:28]: こちらは、ファイルが存在しなかった場合はどうなりますでしょうか？ エラーにはならずに空の内容になる感じでしょうか。

**Review by piroor — COMMENTED** [review]
一部パッと判断が付かない箇所があったので、コメントを頂けましたら幸いです！

File: Dialog/ConfigDialog.xaml.cs (hunk 2 above)
Comment (HashidaTKS) [reply Dialog/ConfigDialog.xaml.cs:28]: エラーにはならず、ファイルが見つからなかった旨のログが出て、デフォルト設定（明示的に変更されたパラメータなし）の`Config`オブジェクトが作成されます。
マージ処理は明示的に変更したパラメータについてのみマージを行うため、マージ時にも実質的に何も影響しません。

File: Dialog/ConfigDialog.xaml.cs (hunk 1 above)
Comment (HashidaTKS) [reply Dialog/ConfigDialog.xaml.cs]: 意図せず改行コードが違っていました。修正しました。
（全体的に改行コードが混在しているので、どこかのタイミングで直したいですね。。。）

File: Dialog/ConfigDialog.xaml.cs (hunk 2 above)
Comment (piroor) [reply Dialog/ConfigDialog.xaml.cs:28]: 了解です！

**Review by piroor — APPROVED** [review]
Looks good to merge.

**Pull request merged**